python/cudf/cudf/_lib/scalar.pyx

```diff
@@ -97,61 +86,61 @@ cdef class DeviceScalar:
             A NumPy dtype.
         """
```

Comment:
Is the description of `value` still up to date with respect to handling of nulls? I don't see where we fail at this point for `None` without a `dtype` also being passed.

Reply:
It is, yes. Now the failure first happens at the C++ level, see https://github.com/rapidsai/cudf/blob/branch-23.10/cpp/src/interop/from_arrow.cu#L520. However the old failure mode in Python also still exists. I think this docstring was never quite right, because the failure happens at the Scalar layer before it ever reaches DeviceScalar: https://github.com/rapidsai/cudf/blob/branch-23.10/python/cudf/cudf/core/scalar.py#L237